src/main/java/teammates/ui/controller/InstructorCourseInstructorDeleteAction.java

```diff
@@ -27,6 +28,15 @@ protected ActionResult execute() {
 
         /* Process deleting an instructor and setup status to be shown to user and admin */
         if (hasAlternativeInstructor(courseId, instructorEmail)) {
+            List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+            isError = !logic.isAtLeastOneInstructorVisibleToStudents(instructors, instructorEmail);
```

Comment:
No, i am fetching all the instructor list and check if any one of instructor have visibility true(Other then current instructor) so to check whether any instructor having visibility to student is true i have to fetch all the instructor? 
Correct me if i am going wrong?